### PR TITLE
File chooser

### PR DESCRIPTION
**Pull request by Guiorgy**
Added the ability to create new folders and delete files on the go as requested on #8.

PS. The option to delete is in case the user made a mistake while naming, so they can recreate the folder, plus they may want to delete something else anyways (kind of killing two birds with one stone?)

### CODE REVIEW

**Comment by hedzr** [comment]
wow, I'll check them asap. Thx a lot.

**Comment by Guiorgy** [comment]
hey @hedzr, I also tried to make it possible to select multiple. you can check the 'select-multiple-2' branch in my fork (just run the demo). should I send a pull request, if you are interested of course?

**Comment by hedzr** [comment]
in checking. you did an amaze thing.
and, i'd seen the new function about `select-multiple-2`, i'm waiting for any new PR but after the one getting done.

**Comment by Guiorgy** [comment]
@hedzr,
I have 2 options for you:

'select-multiple-2': allows to select multiple _files_, but **NOT** _directories_. Also, it's up to the developer to track which file was selected, and which deselected.

'select-multiple-3': allows to select both multiple files and folders, and the adapter stores the selected files, but had to do more changes and create a new functional interface 'OnSelectListener'.

basically choose your poison, and I will send the 1 you want :)

PS. also should I revert the demo to what it was?

**Comment by Guiorgy** [comment]
@hedzr, also if you think anything should (or shouldn't) be changed, please let me know.

**Comment by hedzr** [comment]
it's hard to make the choice! blue or red?
we knew that the end-users do choose multiple files in most cases, we supposed. so, perhaps `select-multiple-2` is right?

don't worry the demo, i'm planning to bump to new version and re-release the app to play store.
thank you for your time.
